src/yaml.cpp

```diff
@@ -105,31 +103,66 @@ estd::result<YAML::Node, estd::error> readYamlFile(std::string const & path) {
 	return YAML::Load(buffer.str());
 }
 
-YamlResult<void> mergeYamlNodes(YAML::Node & map_a, YAML::Node map_b) {
-	// Check if the arguments are maps.
-	if (!map_a.IsMap() && !map_a.IsNull()) {
-		return YamlError{"tried to merge into a YAML node that is not a map"};
-	}
-	if (!map_b.IsMap() && !map_b.IsNull()) {
-		return YamlError{"tried to merge from a YAML node that is not a map"};
-	}
-
+// Merge Yaml Nodes (Type: Map).
+YamlResult<void> mergeYamlMaps(YAML::Node & map_a, YAML::Node map_b) {
 	for (YAML::const_iterator iterator = map_b.begin(); iterator != map_b.end(); iterator++) {
 		std::string key = iterator->first.as<std::string>();
 		YAML::Node value = iterator->second;
-		if (value.IsMap() && map_a[key]) {
-			auto merged = mergeYamlNodes(map_a[key], value);
+
+		if (value.IsMap() && map_a[key] && map_a[key].IsMap()) {
+			auto merged = mergeYamlMaps(map_a[key], value);
 			if (!merged) {
 				merged.error().appendTrace({key, "", YAML::NodeType::Map});
 				return merged;
 			}
 			continue;
 		}
+		else if (value.Tag() == "!ordered_dict" && map_a[key]) {
+			auto merged = mergeYamlOrderedDict(map_a[key], value);
+			if(!merged) {
+				merged.error().appendTrace({key, "", YAML::NodeType::Map});
+				return merged;
+			}
+			continue;
+		}
 		map_a[key] = value;
 	}
 	return estd::in_place_valid;
 }
 
+// Merge Yaml Nodes (Type: Ordered Dictionary).
+YamlResult<void> mergeYamlOrderedDict(YAML::Node & map_a, YAML::Node map_b) {
+	for (std::size_t i = 0; i < map_b.size(); i++) {
+		for (std::size_t j = 0; j < map_a.size(); j++){
+			if (map_b[i].IsMap()){
+				mergeYamlMaps(map_a[j], map_b[i]);
+			}
+			else {
+				return YamlError(fmt::format("{} is not a Map. Ordered dictionaries should be a collection of maps", map_b[i]));
+			}
+		}
+	}
+	return estd::in_place_valid;
+}
+
+// Merge Yaml Nodes.
+YamlResult<void> mergeYamlNodes(YAML::Node & map_a, YAML::Node map_b) {
+	if (map_b.IsMap()){
+		mergeYamlMaps(map_a, map_b);
+	}
+	else if (map_b.Tag() == "!ordered_dict") {
+		mergeYamlOrderedDict(map_a, map_b);
+	}
+	else if (map_b.IsNull()){
+		return estd::in_place_valid;
+	}
+	else {
+		return YamlError("map_b is neither a Map nor an Ordered dictionary");
+	}
+	return estd::in_place_valid;
```

Comment:
Same as before: we should check the type/tag on `node_a`, not on `node_b` to determine what we want to do. And then the type/structure of `node_b` still needs to be validated, but that makes more sense to do in the specialized functions.

Reply:
Please also make sure to follow the Fizyr style guide, so no line breaks between `}` and `else`.